tests/performance/density-kwok.go

```diff
@@ -0,0 +1,305 @@
+	defaultVMCount      = 1000
+)
+
+var (
+	nodeCount = getNodeCount()
+	vmCount   = getVMCount()
+)
+
+var _ = SIGDescribe("[KWOK] Control Plane Performance Density Testing using kwok", func() {
+	var (
+		kubevirtClient kubecli.KubevirtClient
+		k8sClient      *kubernetes.Clientset
+		startTime      time.Time
+	)
+
+	artifactsDir, _ := os.LookupEnv("ARTIFACTS")
+
+	BeforeEach(func() {
+		skipIfNoKWOKPerformanceTests()
+		startTime = time.Now()
+	})
+
+	Describe("kwok density tests", func() {
+		Context(fmt.Sprintf("create a batch of %d fake Nodes", nodeCount), func() {
+			It("should successfully create fake nodes", func() {
+				kubevirtClient = kubevirt.Client()
+
+				config, err := kubecli.GetKubevirtClientConfig()
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "failed to get client config: %v\n", err)
+					return
+				}
+
+				k8sClient, err = kubernetes.NewForConfig(config)
+				if err != nil {
+					fmt.Fprintf(os.Stderr, "failed to create k8s client: %v\n", err)
+					panic(err)
+				}
+
+				By("create fake Nodes")
+				createFakeNodesWithKwok(k8sClient, nodeCount)
+
+				By("Get the list of nodes")
+				_, err = k8sClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+				if err != nil {
+					log.Fatalf("Failed to list nodes: %v", err)
+				}
+			})
+		})
+
+		Context(fmt.Sprintf("create a batch of %d fake VMIs", vmCount), func() {
+			It("should sucessfully create all fake VMIs", func() {
+				By("Creating a batch of fake VMIs")
+				createFakeVMIBatchWithKWOK(kubevirtClient, vmCount)
+
+				By("Waiting for a batch of VMIs")
+				waitRunningVMI(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				By("Deleting fake VMIs")
+				deleteAndVerifyFakeVMIBatch(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				By("Collecting metrics")
+				collectMetrics(startTime, filepath.Join(artifactsDir, "VMI-kwok-perf-audit-results.json"))
+
+				//wait for 5 mins to bring the metrics to steady state
+				time.Sleep(5 * time.Minute)
+			})
+		})
+
+		Context(fmt.Sprintf("create a batch of %d fake VMs", vmCount), func() {
+			It("should sucessfully create all fake VMs", func() {
+				By("Creating a batch of VMs")
+				createFakeBatchRunningVMWithKwok(kubevirtClient, vmCount)
+
+				By("Waiting for a batch of VMs")
+				waitRunningVMI(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				By("Deleting fake VMs")
+				deleteAndVerifyFakeVMBatch(kubevirtClient, vmCount, vmBatchStartupLimit)
+
+				By("Collecting metrics")
+				collectMetrics(startTime, filepath.Join(artifactsDir, "VM-kwok-perf-audit-results.json"))
+			})
+		})
+
+		Context("Deleting fake nodes", func() {
+			It("Successfully delete fake nodes", func() {
+				for i := 1; i <= nodeCount; i++ {
+					nodeName := fmt.Sprintf("kwok-node-%d", i)
+					err := k8sClient.CoreV1().Nodes().Delete(context.TODO(), nodeName, metav1.DeleteOptions{})
+					if err != nil {
+						log.Fatalf("Failed to delete node %s: %v", nodeName, err)
+					}
+				}
+			})
+		})
```

Comment:
Where are these nodes created? 
Are these context executed ordered?

Reply:
> Where are these nodes created?
@fossedihelm these nodes are created on k8s control plane but not tied with the real hardware. These are visible same like the real nodes on the k8s cluster via `kubectl get nodes`

> Are these context executed ordered?
Yes, these are run serially, we are making use of SIGDescribe https://github.com/kubevirt/kubevirt/blob/f1da4887145c27fc88c850c2479bc7a2caf2637b/tests/performance/framework.go#L57

Reply:
AFAIK Serial does not guarantee the execution order.
If you want to guarantee the order you should use `Ordered`